chaos_monkey/network/network_scanner.py

```diff
@@ -27,48 +52,72 @@ def initialize(self):
         LOG.info("Found local IP addresses of the machine: %r", self._ip_addresses)
         # for fixed range, only scan once.
         if WormConfiguration.range_class is FixedRange:
-            self._ranges = [WormConfiguration.range_class(None)]
+            self._ranges = [WormConfiguration.range_class(fixed_addresses=WormConfiguration.range_fixed)]
         else:
             self._ranges = [WormConfiguration.range_class(ip_address)
                             for ip_address in self._ip_addresses]
         if WormConfiguration.local_network_scan:
             self._ranges += [FixedRange([ip_address for ip_address in get_ips_from_interfaces()])]
         LOG.info("Base local networks to scan are: %r", self._ranges)
 
-    def get_victim_machines(self, scan_type, max_find=5, stop_callback=None):
-        assert issubclass(scan_type, HostScanner)
+    def get_victim_machines(self, scan_type, max_find=5, scan_size=ITERATION_BLOCK_SIZE, stop_callback=None):
```

Comment:
Looks good. Though I feel the full potential of these tools are not utilized.
Say we have 20 machines to scan. What currently happens is it will scan 5 of them in parallel, and yield them on request. Meanwhile, when the main thread deals with the results of the first 5 machines, we are not scanning the other 15.
I'm sure we can think of a way to solve this